flux-typeck/src/type_env/paths_tree.rs

```diff
@@ -152,6 +152,7 @@ impl PathsTree {
                     }
                     PlaceElem::Deref => {
                         let ty = node.expect_ty();
+                        let ty = ty.unconstr();
```

Comment:
Was this failing in one of the examples you tried? I feel it should be covered by a call to `unpack` somewhere else. Everything going on in this file is very subtle and I'm not entirely sure if this line is sound so I'd prefer to remove it if it's not strictly necessary for the examples we have so far.

Reply:
Aha, so yes this _was_ failing a test which necessitated this code, BUT it turns out that was _before_ I handled the `unpack` properly, and now in fact, it is not needed. good catch!